numpy/core/tests/test_numeric.py

```diff
@@ -1480,7 +1480,9 @@ def test_simple_int32_inout(self):
         M = np.float64(2)
         ac = np.zeros(a.shape, dtype=np.int32)
         act = ac.copy()
-        self.fastclip(a, m, M, ac)
+        with assert_warns(DeprecationWarning):
+            # NumPy 1.17.0, 2018-02-24 - casting is unsafe
+            self.fastclip(a, m, M, ac)
```

Comment:
Would it make sense to add the `casting='unsafe'` version and check it gives no warnings?

Reply:
Probably, but I ran out of energy for more tests here

Reply:
I've parametrized this unit test to check for both `casting=None` and `casting="unsafe"` scenarios, as requested. As expected, the latter does indeed suppress the warning.